avogadro/core/residuedata.h

```diff
@@ -0,0 +1,110 @@
+#ifndef AVOGADRO_CORE_RESIDUE_DATA
+#define AVOGADRO_CORE_RESIDUE_DATA
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace Avogadro {
+namespace Core {
+
+class ResidueData
+{
+private:
+  std::string m_residueName;
+  std::vector<std::string> m_residueAtomNames;
+  std::vector<std::pair<std::string, std::string>> m_residueSingleBonds;
+  std::vector<std::pair<std::string, std::string>> m_residueDoubleBonds;
+
+public:
+  ResidueData() {}
+  ResidueData(std::string name, std::vector<std::string> atomNames,
+              std::vector<std::pair<std::string, std::string>> singleBonds,
+              std::vector<std::pair<std::string, std::string>> doubleBonds)
+  {
+    m_residueName = name;
+    m_residueAtomNames = atomNames;
+    m_residueSingleBonds = singleBonds;
+    m_residueDoubleBonds = doubleBonds;
+  }
+
+  ResidueData(const ResidueData& other)
+  {
+    m_residueName = other.m_residueName;
+    m_residueAtomNames = other.m_residueAtomNames;
+    m_residueSingleBonds = other.m_residueSingleBonds;
+    m_residueDoubleBonds = other.m_residueDoubleBonds;
+  }
+
+  ResidueData& operator=(ResidueData other)
+  {
+    using std::swap;
+    swap(*this, other);
+    return *this;
+  }
+
+  std::vector<std::pair<std::string, std::string>> residueSingleBonds()
+  {
+    return m_residueSingleBonds;
+  }
+
+  std::vector<std::pair<std::string, std::string>> residueDoubleBonds()
+  {
+    return m_residueDoubleBonds;
+  }
+};
+
+ResidueData alaData("ALA",
+  // Atoms
+  { "CA", "N", "CB", "C", "O" },
+  // Single bonds
+  { 
+    { "CA", "N" },
+    { "CA", "CB" },
+    { "CA", "C" },
+    { "N", "H" },
+    { "N", "HN" },
+    { "HA", "CA" },
+    { "HB1", "CB" },
+    { "HB2", "CB" },
+    { "HB3", "CB" } 
+  },
+  // Double bonds
+  { { "C", "O" } });
+
+ResidueData leuData("LEU",
+  // Atoms
+  { "CA", "N", "CB", "C", "O", "CG", "CD1", "CD2" },
+  // Single bonds
+  {
+    { "CA", "N" },
+    { "CA", "CB" },
+    { "CA", "C" },
+    { "CB", "CG" },
+    { "CG", "CD1" },
+    { "CG", "CD2" },
+    { "N", "H" },
+    { "N", "HN" },
+    { "HA", "CA" },
+    { "HB1", "CB" },
+    { "HB2", "CB" },
+    { "HD11", "CD1" },
+    { "HD12", "CD1" },
+    { "HD13", "CD1" },
+    { "HD21", "CD2" },
+    { "HD22", "CD2" },
+    { "HD23", "CD2" },
+    { "HG", "CG" },
+  },
+  // Double bonds
+  { { "C", "O" } });
+
```

Comment:
What I'd suggest is to write a short Python script or keyboard macro that processes the resdata.txt from Open Babel into this form and 'just do it'.

Reply:
It's also probably worth saving the script, since I think there's a larger ligand database now that could eventually be processed. (OB includes HOH, HEM, etc.)

Reply:
Sure. Can I save a local copy of resData (or if it is possible to retrieve resData.txt from the OpenBabel prefix, how do I go about) and write a parser to fetch data from the file?

Reply:
No, I think it's better to have code like this - I'm just saying that you can write a script that will automate the syntax you've used here. (Similar to `utilities/bodr/bodrparse.cxx`)